jdaviz/configs/specviz/plugins/parsers.py

```diff
@@ -30,32 +32,52 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
+    elif isinstance(data, Spectrum1D):
+        data = [data]
+        data_label = [data_label]
+    elif isinstance(data, SpectrumList):
+        pass
+    else:
         path = pathlib.Path(data)
 
         if path.is_file():
-            data = Spectrum1D.read(str(path), format=format)
+            try:
+                data = [Spectrum1D.read(str(path), format=format)]
+                data_label = [data_label]
+            except IORegistryError:
+                # Multi-extension files may throw a registry error
+                data = SpectrumList.read(str(path), format=format)
         else:
             raise FileNotFoundError("No such file: " + str(path))
-    # If not, it must be a Spectrum1D object. Otherwise, it's unsupported
-    except TypeError:
-        if type(data) is SpectrumCollection:
-            raise TypeError(
-                "SpectrumCollection detected. Please provide a Spectrum1D"
-            )
-        elif type(data) is not Spectrum1D:
-            raise TypeError("Data is not a Spectrum1D object or compatible file")
+
+    if isinstance(data, SpectrumList):
+        if not isinstance(data_label, (list, tuple)):
+            temp_labels = []
+            for i in range(len(data)):
+                temp_labels.append(f"{data_label} {i}")
+            data_label = temp_labels
+        elif len(data_label) != len(data):
+            raise ValueError(f"Length of data labels list ({len(data_label)}) is different"
+                             f" than length of list of data ({len(data)})")
 
     # If there's already data in the viewer, convert units if needed
     current_spec = app.get_data_from_viewer("spectrum-viewer")
-    if current_spec != {} and current_spec is not None:
-        spec_key = list(current_spec.keys())[0]
-        current_unit = current_spec[spec_key].spectral_axis.unit
-        if data.spectral_axis.unit != current_unit:
-            data = Spectrum1D(flux=data.flux,
-                              spectral_axis=data.spectral_axis.to(current_unit))
-
-    app.add_data(data, data_label)
-    if show_in_viewer:
-        app.add_data_to_viewer("spectrum-viewer", data_label)
+    with app.data_collection.delay_link_manager_update():
+        for i in range(len(data)):
+            spec = data[i]
+            if current_spec != {} and current_spec is not None:
+                spec_key = list(current_spec.keys())[0]
+                current_unit = current_spec[spec_key].spectral_axis.unit
+                if spec.spectral_axis.unit != current_unit:
+                    spec = Spectrum1D(flux=spec.flux,
+                                      spectral_axis=spec.spectral_axis.to(current_unit))
+
+            app.add_data(spec, data_label[i])
+            if show_in_viewer:
+                app.add_data_to_viewer("spectrum-viewer", data_label[i])
+
+            # Only auto-show the first spectrum in a list
+            show_in_viewer = False
```

Comment:
Same result but less lines?

```suggestion
            # Only auto-show the first spectrum in a list
            if i == 0 and show_in_viewer:
                app.add_data_to_viewer("spectrum-viewer", data_label[i])
```

Reply:
Whoops, apparently I can't commit this from here now.